compiler/codegen/OMRLinkage.hpp

```diff
@@ -132,6 +132,41 @@ class OMR_EXTENSIBLE Linkage
     */
    virtual void performPostBinaryEncoding() { }
 
+   /** @brief
+    *    Gets the offset (in number of bytes) from the stack frame pointer to the location on the stack where the first
+    *    (closest to the frame pointer) parameter is located. 
+    *
+    *  @details
+    *    For example given the following stack frame layout, and assuming a call to a function with 4 parameters:
+    *
+    *    @code
+    *        0x0588     +--------+
+    *                   |  ARG1  |
+    *        0x0580     +--------+
+    *                   |  ARG2  |
+    *        0x0578     +--------+
+    *                   |  ARG3  |
+    *        0x0570     +--------+   <- offset to first parm relative to the frame pointer (0x0570 - 0x0530 = 0x0040)
+    *                   |  ARG4  |
+    *        0x0568     +--------+ 
+    *                   |  ....  |
+    *        0x0530     +--------+   <- frame pointer
+    *                   |  ....  |
+    *        0x0500     +--------+   <- stack pointer
+    *    @endcode
+    *
+    *    The offset returned by this function (0x0040 in the above example) may not be the first argument (in argument
+    *    order) passed by the caller on the stack. It is up to the linkage to make use of this function to initialize
+    *    parameter offsets depending on the order in which the caller passes the arguments to the callee.
+    */
+   virtual int32_t getOffsetToFirstParm() const;
+
+   /** @brief
+    *    Sets the offset (in number of bytes) from the stack frame pointer to the location on the stack where the first
+    *    (closest to the frame pointer) parameter is located. 
+    */
+   virtual int32_t setOffsetToFirstParm(int32_t offset);
```

Comment:
Is there a reason that the getter/setter methods here are marked as virtual? I'm not sure I understand what the use case would be for a subclass to override these...

As a side note, if the virtual modifier were dropped, I think the definitions for these methods would be best moved into `OMRLinkage_inlines.hpp` as their implementations are trivial.

Reply:
Agreed. Fixed in a660739